lang/ruby/lib/avro/schema.rb

```diff
@@ -33,7 +35,10 @@ class Schema
     LONG_MAX_VALUE = (1 << 63) - 1
 
     def self.parse(json_string)
-      real_parse(MultiJson.load(json_string), {})
+      json_obj = MultiJson.load(json_string)
+      schema = real_parse(json_obj, {})
+      schema.logical_type = LogicalTypes.parse(json_obj) if json_obj.is_a?(Hash)
```

Comment:
Ok, I'll take a look at changing it next week.
On lør. 5. sep. 2015 at 02.07 Ryan Blue <notifications@github.com> wrote:

> In lang/ruby/lib/avro/schema.rb
> <https://github.com/apache/avro/pull/44#discussion_r38804034>:
>
> > @@ -33,7 +35,10 @@ class Schema
> >      LONG_MAX_VALUE = (1 << 63) - 1
> >
> >      def self.parse(json_string)
> > -      real_parse(MultiJson.load(json_string), {})
> > +      json_obj = MultiJson.load(json_string)
> > +      schema = real_parse(json_obj, {})
> > +      schema.logical_type = LogicalTypes.parse(json_obj) if json_obj.is_a?(Hash)
>
> I see what you're saying about the difficulty of doing it that way, which
> is pretty annoying. But, I think it would be better to avoid exposing the
> accessor.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/avro/pull/44/files#r38804034>.
>